### ISSUE TITLE
'NitroModules/PromiseHolder.hpp' file not found - iOS

### ISSUE THREAD

**Issue by emenocal07** [margelo/react-native-nitro-sqlite#90]
## ISSUE:
Project fails building for iOS:
```
CompileC /Users/ernestoesp/Library/Developer/Xcode/DerivedData/myApp-csrmfklfutoyzkcfxkuurxwaphok/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNNitroSQLite.build/Objects-normal/x86_64/OnLoad.o /Users/ernestoesp/Documents/Projects/imyApp/node_modules/react-native-nitro-sqlite/ios/OnLoad.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'RNNitroSQLite' from project 'Pods')
    cd /Users/ernestoesp/Documents/Projects/myApp/ios/Pods
    
    Using response file: /Users/ernestoesp/Library/Developer/Xcode/DerivedData/myApp-csrmfklfutoyzkcfxkuurxwaphok/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNNitroSQLite.build/Objects-normal/x86_64/af3fcb34312c57c0f52879cdce924b91-common-args.resp
    
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -ivfsstatcache /Users/ernestoesp/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator18.1-22B74-3d93aac3a03ebac1dd8474c5def773dc.sdkstatcache -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -fno-color-diagnostics -fmodules-prune-interval\=86400 -fmodules-prune-after\=345600 -fbuild-session-file\=/Users/ernestoesp/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror\=non-modular-include-in-framework-module -Wno-trigraphs -Wno-missing-field-initializers -Wno-missing-prototypes -Werror\=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror\=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror\=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -Wno-c++11-extensions -Wno-implicit-fallthrough -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -Wno-sign-conversion -Winfinite-recursion -Wmove -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wrange-loop-analysis -Wno-semicolon-before-method-body -Wunguarded-availability -index-store-path /Users/ernestoesp/Library/Developer/Xcode/DerivedData/myApp-csrmfklfutoyzkcfxkuurxwaphok/Index.noindex/DataStore -Wno-shorten-64-to-32 -Wno-comma -Wno-unreachable-code -Wno-conditional-uninitialized -Wno-deprecated-declarations -Wno-comma -Wno-shorten-64-to-32 @/Users/ernestoesp/Library/Developer/Xcode/DerivedData/myApp-csrmfklfutoyzkcfxkuurxwaphok/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNNitroSQLite.build/Objects-normal/x86_64/af3fcb34312c57c0f52879cdce924b91-common-args.resp -DFOLLY_NO_CONFIG -DFOLLY_MOBILE\=1 -DFOLLY_USE_LIBCPP\=1 -DFOLLY_CFG_NO_COROUTINES\=1 -DFOLLY_HAVE_CLOCK_GETTIME\=1 -Wno-comma -Wno-shorten-64-to-32 -include /Users/ernestoesp/Documents/Projects/myApp/ios/Pods/Target\ Support\ Files/RNNitroSQLite/RNNitroSQLite-prefix.pch -MMD -MT dependencies -MF /Users/ernestoesp/Library/Developer/Xcode/DerivedData/myApp-csrmfklfutoyzkcfxkuurxwaphok/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNNitroSQLite.build/Objects-normal/x86_64/OnLoad.d --serialize-diagnostics /Users/ernestoesp/Library/Developer/Xcode/DerivedData/myApp-csrmfklfutoyzkcfxkuurxwaphok/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNNitroSQLite.build/Objects-normal/x86_64/OnLoad.dia -c /Users/ernestoesp/Documents/Projects/myApp/node_modules/react-native-nitro-sqlite/ios/OnLoad.mm -o /Users/ernestoesp/Library/Developer/Xcode/DerivedData/myApp-csrmfklfutoyzkcfxkuurxwaphok/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNNitroSQLite.build/Objects-normal/x86_64/OnLoad.o -index-unit-output-path /Pods.build/Debug-iphonesimulator/RNNitroSQLite.build/Objects-normal/x86_64/OnLoad.o

In file included from /Users/ernestoesp/Documents/Projects/myApp/node_modules/react-native-nitro-sqlite/ios/OnLoad.mm:2:
/Users/ernestoesp/Documents/Projects/myApp/ios/Pods/Headers/Private/RNNitroSQLite/RNNitroSQLite-Swift-Cxx-Umbrella.hpp:23:10: fatal error: 'NitroModules/PromiseHolder.hpp' file not found
   23 | #include <NitroModules/PromiseHolder.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

## CONFIG:
package.json:
``` 
"react-native": "0.75.4",
"react-native-nitro-modules": "^0.17.0",
"react-native-nitro-sqlite": "^9.0.1",
````

Right now using xCode 16.1
Using React Native CLI setup

## EXPECTED BEHAVIOUR:

App compiles successfully on iOS.

Options already tried and issue persist:
- Deintegrating pods and running pod install again
- Removing derived data
- Updating xCode from 15.4 to 16.1

**Comment by chrispader** [comment]
Yes sorry, this is fixed by using `react-native-nitro-modules@^0.16.2`, but not version `v17.0.0` yet. I just published a new version `9.0.2` that enforces this peer dependency version

**Comment by emenocal07** [comment]
Thanks!
